app/services/keycloak_oauth/authorizable_service.rb

```diff
@@ -0,0 +1,24 @@
+require 'net/http'
+
+module KeycloakOauth
+  class AuthorizableError < StandardError; end
+
+  class AuthorizableService
+    HTTP_SUCCESS_CODES = [Net::HTTPOK, Net::HTTPNoContent]
+    DEFAULT_CONTENT_TYPE = 'application/x-www-form-urlencoded'.freeze
+    AUTHORIZATION_HEADER = 'Authorization'.freeze
+
+    private
+
+    def parsed_response(http_response)
+      response = http_response.body.present? ? JSON.parse(http_response.body) : http_response.body
+
+      return response if HTTP_SUCCESS_CODES.include?(http_response.code_type)
+
+      # TODO: For now, we assume that the access token is always valid.
+      # We do not yet handle the case where a refresh token is passed in and
+      # used if the access token has expired.
+      raise KeycloakOauth::AuthorizableError.new(response)
```

Comment:
Did you check what happens if the user is already logged out? For example, does Keycloak return a 401 response code?

Reply:
Hmm I don't think I did. Will check.

Reply:
```ruby
    9:     def logout
   10:       byebug
=> 11:       parsed_response(post_logout)
   12:     end
   13:
   14:     private
   15:
(byebug) post_logout
#<Net::HTTPNoContent 204 No Content readbody=true>
(byebug) post_logout
#<Net::HTTPNoContent 204 No Content readbody=true>
```

Looks like it returns a 200 even if the user is logged out already (and I did double check that the user session had been terminated before calling `post_logout` again.